doc/examples/particle_filtering_fiber_tracking.py

```diff
@@ -0,0 +1,151 @@
+
+
+"""
+CMC/ACT Tissue Classifiers
+---------------------
+Continuous map criterion (CMC) [Girard2014]_ and Anatomically-constrained
+tractography (ACT) [Smith2012]_ both uses PVEs information from
+anatomical images to determine when the tractography stops.
+Both tissue classifiers use a trilinear interpolation
+at the tracking position. CMC tissue classifier uses a probability derived from
+the PVE maps to determine if the streamline reaches a 'valid' or 'invalid'
+region. ACT uses a fixed threshold on the PVE maps. Both tissue classifiers can
+be used in conjunction with PFT. In this example, we used CMC.
+"""
+
+import matplotlib.pyplot as plt
+from dipy.tracking.local import CmcTissueClassifier
+
+voxel_size = np.average(img_pve_wm.get_header()['pixdim'][1:4])
+step_size = 0.2
+
+cmc_classifier = CmcTissueClassifier.from_pve(img_pve_wm.get_data(),
+                                              img_pve_gm.get_data(),
+                                              img_pve_csf.get_data(),
+                                              step_size=step_size,
+                                              average_voxel_size=voxel_size)
+
+# seeds are place in voxel of the corpus callosum containing only white matter
+seed_mask = labels == 2
+seed_mask[img_pve_wm.get_data() < 0.5] = 0
+seeds = utils.seeds_from_mask(seed_mask, density=2, affine=affine)
+
+# Particle Filtering Tractography
+pft_streamlines = ParticleFilteringTracking(dg,
+                                            cmc_classifier,
+                                            seeds,
+                                            affine,
+                                            max_cross=1,
+                                            step_size=step_size,
+                                            maxlen=1000,
+                                            pft_back_tracking_dist=2,
+                                            pft_front_tracking_dist=1,
+                                            particle_count=15,
+                                            return_all=False)
+streamlines = [s for s in pft_streamlines]
+save_trk("pft_streamline.trk", streamlines, affine, shape)
+
+
+renderer.clear()
+renderer.add(actor.line(streamlines, line_colors(streamlines)))
+window.record(renderer, out_path='pft_streamlines.png', size=(600, 600))
+
+"""
+.. figure:: pft_streamlines.png
+ :align: center
+
+ **Particle Filtering Tractography**
+"""
+
+# Local Probabilistic Tractography
+probabilistic_streamlines = LocalTracking(dg,
+                                          cmc_classifier,
+                                          seeds,
+                                          affine,
+                                          max_cross=1,
+                                          step_size=step_size,
+                                          maxlen=1000,
+                                          return_all=False)
+probabilistic_streamlines = [s for s in probabilistic_streamlines]
```

Comment:
This is the old standard for generating streamlines which has memory issues what you should use is 
```
from dipy.tracking.streamline import Streamlines
streamlines = Streamlines(probabilistic_streamlines)
```
See example `tracking_quick_start.py` for more.